### PR TITLE
Extract common Pageable Table component.

### PR DESCRIPTION
**Pull request by bodnarbm**
First pass at extracting out the table component common between a number of management pages. 

Hopefully, we can eventually extract out more of the functionality of these pages into either this PageableTable component or into another ManagePage component. 

Creating these manage pages is pretty common in our workflows and with so much shared code, extracting out the common components should strengthen code reuse and experimentation with new display structures/features.

### CODE REVIEW

**Comment by brianghig** [comment]
Hooray for net negative merge requests (considering the +150 lines in a spec)!

File: src/client/app/access-checker/admin-cache-entries.component.ts [hunk 1]
```diff
@@ -37,8 +38,8 @@ export class AdminCacheEntriesComponent {
 		this.alertService.clearAllAlerts();
 
 		this.pagingOpts = new PagingOptions();
-		this.pagingOpts.sortField = this.sortOpts.key.sortField;
-		this.pagingOpts.sortDir = this.sortOpts.key.sortDir;
+		this.pagingOpts.sortField = this.sortOpts['key'].sortField;
```
Comment (brianghig) [inline src/client/app/access-checker/admin-cache-entries.component.ts]: Why change from dot notation?

File: src/client/app/admin/end-user-agreement/admin-list-euas.component.ts [hunk 2]
```diff
@@ -78,8 +79,8 @@ export class AdminListEuasComponent implements OnInit {
 			this.pagingOpts = cachedFilter.paging;
 		} else {
 			this.pagingOpts = new PagingOptions();
-			this.pagingOpts.sortField = this.sortOpts.title.sortField;
-			this.pagingOpts.sortDir = this.sortOpts.title.sortDir;
+			this.pagingOpts.sortField = this.sortOpts['title'].sortField;
```
Comment (brianghig) [inline src/client/app/admin/end-user-agreement/admin-list-euas.component.ts]: Similarly, why change from dot notation?

**Comment by bodnarbm** [comment]
Rebased after the update to Angular 2.4 and Typescript 2.2.

**Review by brianghig — APPROVED** [review]

File: src/client/app/access-checker/admin-cache-entries.component.ts (hunk 1 above)
Comment (bodnarbm) [reply src/client/app/access-checker/admin-cache-entries.component.ts]: Weird. That change is not in the current version of the branch (Typescript 2.2 allows dot notation for string indexers, while Typescript 2.1) does not. Not sure why the code review showed ['key'] for you.

Comment (bodnarbm) [reply src/client/app/access-checker/admin-cache-entries.component.ts]: Ahh.. the code review was done on an older commit. That's why.

**Review by ymao1 — APPROVED** [review]

**Pull request merged**